apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+    subjectLengthNotice: null
+  },
+
+  multiSimActionButton: null,
+  init: function conv_init() {
+    var templateIds = [
+      'message',
+      'message-sim-information',
+      'message-status',
+      'not-downloaded',
+      'recipient',
+      'date-group',
+      'header',
+      'group-header'
+    ];
+
+    // Fields with 'messages' label
+    [
+      'container', 'to-field', 'recipients-list', 'compose-form', 'header',
+      'edit-header', 'check-uncheck-all-button', 'contact-pick-button',
+      'send-button', 'delete-button', 'call-number-button', 'options-button',
+      'new-message-notice', 'edit-mode', 'edit-form', 'header-text',
+      'max-length-notice', 'convert-notice', 'resize-notice',
+      'new-message-notice', 'subject-max-length-notice', 'sms-counter-notice',
+      'recipient-suggestions'
+    ].forEach(function(id) {
+      this[Utils.camelCase(id)] = document.getElementById('messages-' + id);
+    }, this);
+
+    this.mainWrapper = document.getElementById('main-wrapper');
+    this.threadMessages = document.getElementById('thread-messages');
+
+    window.addEventListener('resize', this.resizeHandler.bind(this));
+
+    // binding so that we can remove this listener later
+    this.onVisibilityChange = this.onVisibilityChange.bind(this);
+    document.addEventListener('visibilitychange',
+                              this.onVisibilityChange);
+
+    this.toField.addEventListener(
+      'keypress', this.toFieldKeypress.bind(this), true
+    );
+
+    this.toField.addEventListener(
+      'input', this.toFieldInput.bind(this), true
+    );
+
+    this.toField.addEventListener(
+      'focus', this.toFieldInput.bind(this), true
+    );
+
+    this.sendButton.addEventListener(
+      'click', this.onSendClick.bind(this)
+    );
+
+    this.sendButton.addEventListener(
+      'contextmenu', this.onSendClick.bind(this)
+    );
+
+    this.container.addEventListener(
+      'scroll', this.manageScroll.bind(this)
+    );
+
+    this.editHeader.addEventListener(
+      'action', this.cancelEdit.bind(this)
+    );
+
+    this.header.addEventListener(
+      'action', this.onHeaderAction.bind(this)
+    );
+
+    this.optionsButton.addEventListener(
+      'click', this.showOptions.bind(this)
+    );
+
+    this.callNumberButton.addEventListener('click', function() {
+      ActivityPicker.dial(Threads.active.participants[0]);
+    });
+
+    this.deleteButton.addEventListener(
+      'click', this.delete.bind(this)
+    );
+
+    this.headerText.addEventListener(
+      'click', this.onHeaderActivation.bind(this)
+    );
+
+    this.newMessageNotice.addEventListener(
+      'click', this.onNewMessageNoticeClick.bind(this)
+    );
+
+    // These events will be handled in handleEvent function
+    this.container.addEventListener('click', this);
+    this.container.addEventListener('contextmenu', this);
+    this.editForm.addEventListener('submit', this);
+    this.composeForm.addEventListener('submit', this);
```

Comment:
> actually, to prevent submit, we just need a <button type="button">

Need to check if we have something specific about ```[type=submit]``` or ```[type=button]```` in BB, if not - then it's good.

 > But more importantly, for accessibility, maybe we'd like the "submit" event to actually run the action (instead of listening for the click event on a button)

We don't directly listen for ```click``` event on send button we listen for ```mulisim action callback``` instead...